src/tools/kata-ctl/src/types.rs

```diff
@@ -33,3 +34,28 @@ pub struct CheckItem<'a> {
     pub fp: BuiltinCmdFp,
     pub perm: PermissionType,
 }
+
+// Builtin module parameter check handler type
+pub type BuiltinModuleParamFp = fn(args: &str) -> bool;
+
+// KernelParamType encodes the value and handler function
+// for kernel module parameters
+#[derive(Clone)]
+pub enum KernelParamType<'a> {
+    Simple(&'a str),
+    Predicate(BuiltinModuleParamFp),
+}
+
+#[derive(Clone)]
+pub struct KernelParam<'a> {
+    pub name: &'a str,
+    pub value: KernelParamType<'a>,
+}
+
+// KernelModule is used to describe a kernel module along with its required parameters.
+// Parameters is a map,value structure
+// #[derive(Clone, Debug)]
+pub struct KernelModule<'a> {
+    pub name: &'a str,
+    pub parameter: KernelParam<'a>,
```

Comment:
This is fine for now. There is a chance in the future that we may need to check >1 parameters for a single module. If and when that happens, we should probably make this a `Vec<KernelParam<'a>>`.

Reply:
@jodh-intel  thank you for the suggestion